includes/integrations/class.llms.integration.buddypress.php

```diff
@@ -1,36 +1,60 @@
 <?php
 /**
- * BuddyPress Integration
+ * BuddyPress Integration.
  *
  * @package LifterLMS/Integrations/Classes
  *
  * @since 1.0.0
- * @version 3.37.17
+ * @version [version]
  */
 
 defined( 'ABSPATH' ) || exit;
 
 /**
- * BuddyPress Integration
+ * BuddyPress Integration.
  *
  * @since 1.0.0
- * @since 3.12.2 Unknown.
- * @since 3.14.4 Unknown.
  * @since 3.37.17 Fixed `courses` pagination.
  */
 class LLMS_Integration_Buddypress extends LLMS_Abstract_Integration {
 
 	public $id = 'buddypress';
 
 	/**
-	 * Display order on Integrations tab
+	 * Display order on Integrations tab.
 	 *
 	 * @var integer
 	 */
 	protected $priority = 5;
 
 	/**
-	 * Configure the integration
+	 * Current endpoint's key being processed.
+	 *
+	 * @var string
+	 */
+	private $current_endpoint_key;
```

Comment:
I usually prefer to make properties `protected` so that a class that extends this class can change it. Another option is to add getter and setter methods.

Reply:
I recently started to make props and methods `private` as much as possible, mostly because if we want to turn a property/method to protected/public because a dev requested it, or because we need it to be so, we can do it without introduce bc (backward compatibility) issues, while the opposite, of course, it will introduce a bc issue.
I'd say `protected` would be fine as well, but given the logic here I doubt either making it `protected` or introducing `protected` getters/setters would be of any help for an extending class.
Use case?

Reply:
It doesn't seem likely that a 3rd party would extend this class, so the visibility on these two properties probably don't matter.

However, I get frustrated a few times a year when the project I'm working on needs access to a variable that is private and does not have a getter/setter, or I am extending a class and can't override a private method.

I see you point about backward compatibility though. I need to read up on the pros and cons of private/protected to help balance future decisions.

Reply:
I go back and forth on this... however... as a maintainer of a codebase I more frequently bang my head against a wall trying to figure out *how to deprecate something gracefully* than someone complains about something being "private"

I have seen the argument about *never* using private and making everything protected all the time but backwards compat is such a pain and it's so nice when we can simply delete a method we don't need anymore.